main.go

```diff
@@ -261,6 +265,74 @@ func writeNewXWithOptionsAndDefaults(buf *jen.File, c Config) {
 	})
 }
 
+var (
+	DebugMapFieldTag = "debugmap"
+	SensitiveNames   = []string{"secure"}
+)
+
+func writeDebugMap(buf *jen.File, st *types.Struct, c Config) {
+	newFuncName := fmt.Sprintf("DebugMap")
+
+	buf.Comment(fmt.Sprintf("%s returns a map form of %s for debugging", newFuncName, c.TargetTypeName))
+	buf.Func().Params(jen.Id(c.ReceiverId).Id(c.StructName)).Id(newFuncName).Params().Id("map[string]any").BlockFunc(func(grp *jen.Group) {
+		mapId := "debugMap"
+		grp.Id(mapId).Op(":=").Map(jen.String()).Any().Values()
+
+		for i := 0; i < st.NumFields(); i++ {
+			f := st.Field(i)
+			if f.Anonymous() || !f.Exported() {
+				continue
+			}
+
+			tags, err := structtag.Parse(st.Tag(i))
+			if err != nil {
+				panic(err)
+			}
+
+			tag, err := tags.Get(DebugMapFieldTag)
+			if err != nil {
+				fmt.Printf("missing debugmap tag on field %s in type %s\n", f.Name(), c.TargetTypeName)
+				os.Exit(1)
+			}
+
+			switch tag.Value() {
```

Comment:
Do we need `structtag`? Can't you just do:

```suggestion
			switch f.Tag.Get(DebugMapFieldTag) {
```

Reply:
![Screen Shot 2023-05-08 at 4 30 39 PM](https://user-images.githubusercontent.com/4073002/236927909-fcc77964-50b3-4f25-b684-eb949ddd71f1.png)

Reply:
![Screen Shot 2023-05-08 at 4 31 00 PM](https://user-images.githubusercontent.com/4073002/236927970-858f5012-0212-4d23-b238-5eef6cdc77c7.png)

Reply:
Also doesn't work on `st.Tag`, which is just a string